tests/rptest/tests/log_compaction_test.py

```diff
@@ -628,3 +630,153 @@ def seen_compacted_segments():
             self._rpk_client.cluster_config_set(
                 "log_compaction_use_sliding_window", next_sliding_window_config
             )
+
+
+class LogCompactionTxRemovalTest(LogCompactionTestBase, PreallocNodesTest):
+    def __init__(self, test_context):
+        self.test_context = test_context
+        # Run with small segments and a very frequent compaction interval.
+        self.extra_rp_conf = {
+            "log_compaction_interval_ms": 1000,
+            "log_segment_size": 2 * 1024**2,  # 2 MiB
+            "compacted_log_segment_size": 1024**2,  # 1 MiB
+        }
+
+        super().__init__(
+            test_context=test_context,
+            num_brokers=1,
+            node_prealloc_count=1,
+            extra_rp_conf=self.extra_rp_conf,
+        )
+
+        self._rpk_client = RpkTool(self.redpanda)
+
+    def produce(self, test_case):
+        producer = KgoVerifierProducer(
+            context=self.test_context,
+            redpanda=self.redpanda,
+            topic=self.topic_spec.name,
+            use_transactions=True,
+            msg_size=test_case.msg_size,
+            msg_count=test_case.msg_count,
+            transaction_abort_rate=test_case.abort_rate,
+            msgs_per_transaction=test_case.msgs_per_transaction,
+            custom_node=self.preallocated_nodes,
+        )
+
+        producer.start()
+        producer.wait(timeout_sec=180)
+        producer.stop()
+
+    def check_tx_batches(self):
+        viewer = OfflineLogViewer(self.redpanda)
+        assert len(self.redpanda.nodes) == 1
+        node = self.redpanda.nodes[0]
+        num_control_batches = 0
+        num_fence_batches = 0
+        records_and_batches = viewer.read_kafka_records(node, self.topic_spec.name)
+        for record_or_batch in records_and_batches:
+            if "expanded_attrs" not in record_or_batch:
+                continue
+            if record_or_batch["expanded_attrs"]["control_batch"]:
+                num_control_batches += 1
+            if record_or_batch["type_name"] == "tx_fence":
+                num_fence_batches += 1
+
+        assert num_control_batches == 0, (
```

Comment:
I'm wondering how this is possible with the default delete retention ms (the test doesn't seem to override this in rp_conf).